src/AutoRest.CSharp.V3/Generation/Types/TypeFactory.cs

```diff
@@ -186,37 +186,45 @@ public bool TryCreateType(ITypeSymbol symbol, [NotNullWhen(true)] out CSharpType
                 throw new InvalidCastException($"Unexpected type {symbol}");
             }
 
-            bool nullable;
             if (namedTypeSymbol.ConstructedFrom.SpecialType == SpecialType.System_Nullable_T)
             {
-                nullable = true;
-                namedTypeSymbol = (INamedTypeSymbol)namedTypeSymbol.TypeArguments[0];
-            }
-            else
-            {
-                nullable = symbol.NullableAnnotation != NullableAnnotation.NotAnnotated;
+                type = CreateType(namedTypeSymbol.TypeArguments[0]).WithNullable(true);
+                return true;
             }
 
             var fullMetadataName = GetFullMetadataName(namedTypeSymbol);
             var fullyQualifiedName = $"{fullMetadataName}, {namedTypeSymbol.ContainingAssembly.Name}";
             var existingType = Type.GetType(fullMetadataName) ?? Type.GetType(fullyQualifiedName);
+
             if (existingType != null)
             {
                 var arguments = namedTypeSymbol.TypeArguments.Select(a => CreateType(a)).ToArray();
-                type = new CSharpType(existingType, nullable, arguments);
-                return true;
+                type = new CSharpType(existingType, false, arguments);
```

Comment:
What if the existing type is nullable, though? Or is that reserved for when you actually support `x-nullable`?

Reply:
Yes, we assign nullability later in the code.